src/validators/validateCreateNotificationParams.ts

```diff
@@ -6,6 +6,11 @@ export const validateCreateNotificationParams = (
 ): Joi.ValidationResult => Joi.object().keys({
   notificationType: Joi.string().required(),
   toUserId: Joi.number().integer().min(1).required(),
+  context: Joi.object().keys({
+  }).pattern(
+    Joi.string().invalid('from', 'notification', 'message_type', 'toUserId', 'notificationType').regex(/(google).*|(gcm).*/i, { invert: true }).insensitive(),
```

Comment:
The regular expression is matching on the substrings `google` and `gcm` in any position, but we only need to prevent keys that start with those values. To do so, we need to match the start-of-string anchor, `^`. We also don't need to match subsequent characters with `.*`, since we don't need to consume the entire string.

```suggestion
    Joi.string().invalid('from', 'notification', 'message_type', 'toUserId', 'notificationType').regex(/^(google|gcm)/i, { invert: true }).insensitive(),
```

I would still like to extract the forbidden words list; even with the suggested change, this line is more than 150 characters long.